### PR TITLE
EVP Fix

### PR DESCRIPTION
**Pull request by ejohnstown**
There are some cases when the EVP wrapper code could call strncmp with a null pointer. This was refactored to remove this possibility. The main changes were to make the constant pointers to the EVP names strings constant fixed arrays. The references to these checking if these strings were set could be removed as they'd always be present. (ZD10972)

### CODE REVIEW

File: wolfcrypt/src/evp.c [hunk 1]
```diff
@@ -2649,16 +2663,19 @@ static const struct cipher{
...
     #ifdef WOLFSSL_AES_128
-    {AES_128_CBC_TYPE, "AES-128-CBC", NID_aes_128_cbc},
+    {AES_128_CBC_TYPE, EVP_AES_128_CBC, NID_aes_128_cbc},
     #endif
     #ifdef WOLFSSL_AES_192
-    {AES_192_CBC_TYPE, "AES-192-CBC", NID_aes_192_cbc},
+    {AES_192_CBC_TYPE, EVP_AES_192_CBC, NID_aes_192_cbc},
     #endif
     #ifdef WOLFSSL_AES_256
     {AES_256_CBC_TYPE, "AES-256-CBC", NID_aes_256_cbc},
```
Comment (dgarske) [inline wolfcrypt/src/evp.c:2674]: Why not refactor the remainder of these to use the static const char values above?

Example:
```
- {AES_256_CBC_TYPE, "AES-256-CBC", NID_aes_256_cbc},
+ {AES_256_CBC_TYPE, EVP_AES_256_CBC, NID_aes_256_cbc},
```

Etc... below?

File: wolfcrypt/src/evp.c [hunk 2]
```diff
@@ -2817,16 +2846,16 @@ const WOLFSSL_EVP_CIPHER *wolfSSL_EVP_get_cipherbyname(const char *name)
...
-        {"AES-192-CBC", "AES192-CBC"},
-        {"AES-192-CBC", "aes192-cbc"},
+        {EVP_AES_192_CBC, "AES192-CBC"},
+        {EVP_AES_192_CBC, "aes192-cbc"},
         #endif
         #ifdef WOLFSSL_AES_256
-        {"AES-256-CBC", "AES256-CBC"},
-        {"AES-256-CBC", "aes256-cbc"},
+        {EVP_AES_256_CBC, "AES256-CBC"},
+        {EVP_AES_256_CBC, "aes256-cbc"},
```
Comment (dgarske) [inline wolfcrypt/src/evp.c:2858]: Also below here? `EVP_AES_128_ECB` etc...

**Review by dgarske — CHANGES_REQUESTED** [review]
Excellent cleanup! Much better! There are some more items that did not get cleaned up... Seems like an easy finish to get it 100%.

File: wolfcrypt/src/evp.c (hunk 1 above)
Comment (ejohnstown) [reply wolfcrypt/src/evp.c:2674]: Oversight.

**Review by dgarske — APPROVED** [review]
Thanks John! Excellent cleanup!

**Pull request merged**